TELCODOCS-1670: Network matrix flow (#69720)
openshift/openshift-docs: modules/network-flow-matrix.adoc

// Module included in the following assemblies:
//
// * installing/install_config/configuring-firewall.adoc

:_mod-docs-content-type: REFERENCE
[id="network-flow-matrix_{context}"]
= {product-title} network flow matrix

The network flow matrix describes the ingress flows to {product-title} services.
Use the information in the network flow matrix to help you manage ingress traffic.

To view or download the raw CSV content, see link:https://raw.githubusercontent.com/rohennes/openshift-docs/TELCODOCS-1670/snippets/network-flow-matrix.csv[this resource].

[NOTE]
====

> The network flow matrix describes common ingress traffic flows. It does not describe every possible flow. Some flows that are not described in this matrix might be critical to operation. For example, if you block all traffic and then selectively open only the flows described here, you might unintentionally block a necessary flow.

Reviewer: I dont think this note is accurate.
the whole plan is to support this matrix
so it is accurate for base openshift.
what we do want to say, is that it does not encompass optional component (e.g. operators from market place)

Author: I've updated the content accordingly

// Module included in the following assemblies:
//
// * installing/install_config/configuring-firewall.adoc

:_mod-docs-content-type: REFERENCE
[id="network-flow-matrix_{context}"]
= {product-title} network flow matrix

The network flow matrix describes the ingress flows to {product-title} services.
The network information in the matrix is accurate for both bare-metal and cloud environments.
Use the information in the network flow matrix to help you manage ingress traffic.

> You can restrict ingress traffic to essential flows only to improve network security.

Reviewer 2: ```suggestion
You can restrict ingress traffic to only essential flows to improve network security.
```
just a suggestion as the way it reads is somehow awkward

Author: Thanks - actually I just removed "only" and I think the meaning remains the same

> To view or download the raw CSV content, see link:https://raw.githubusercontent.com/rohennes/openshift-docs/TELCODOCS-1670/snippets/network-flow-matrix.csv[this resource].

Reviewer 2: I don't know if Telco is subject to the requirement about [not linking to github repos](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-to-external-websites) - I know some teams allow it, but I thought I'd ask.

Author: Thanks Kerry. This is autogenerated content that we would like to embed directly using the [rawgithub usercontent method](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#embedding-an-external-file) in 4.17. For now, the automation of this is not fully complete - we need to edit a few fields by hand - so this is a halfway house that will be updated in 4.17. Hope that works.

Additionally, consider the following dynamic port ranges when managing ingress traffic:

> * `9000-9999`: Host level services.

Reviewer 2: IBM Style recommends no punctuation for bulleted lists that aren't sentences

> * `9000-9999`: General use ports for system services.

Reviewer: Host level services
(instead of 'General use ports for system services')

the term "general use" might be confusing. I dont think it's wise to use these outside the platform itself

Author: Thanks Yuval, I updated he PR accordingly

* `3000-32767`: Kubernetes node ports.

> * `49152-65535`: Remote procedure call ports.

Reviewer: this is the default, well agreed, dynamic ports range.
not sure if we want to list it at all.
but say we do
I suggest we call it "Dynamic/private port range"

[NOTE]
====

> The network flow matrix describes ingress traffic flows for a base {product-title} installation. It does not describe network flows for additional components, such as optional Operators available from the Red Hat Marketplace. The matrix does not apply for HyperShift, MircoShift, or standalone clusters.

Reviewer 3: ```suggestion
The network flow matrix describes ingress traffic flows for a base {product-title} installation. It does not describe network flows for additional components, such as optional Operators available from the Red Hat Marketplace. The matrix does not apply for Hosted-Control-Plane, MicroShift, or standalone clusters.
```
"HyperShift" is not the RH-approved term, it is Hosted Control Planes. Using the hyphens here because it is a compound modifier.

> The network flow matrix describes ingress traffic flows for a base {product-title} installation. It does not describe network flows for additional components, such as optional Operators available from the Red Hat Marketplace. The matrix does not apply for HyperShift, MircoShift or standalone clusters.

Reviewer 2: ```suggestion
The network flow matrix describes ingress traffic flows for a base {product-title} installation. It does not describe network flows for additional components, such as optional Operators available from the Red Hat Marketplace. The matrix does not apply for HyperShift, MicroShift, or standalone clusters.
```
